sandy/core/endf6.py

```diff
@@ -2088,3 +2089,167 @@ def get_gendf(self,
+                    mat1 = mat
+                logging.debug("\treading subsection MAT1={}/MT1={}"
+                              .format(mat1, mt1))
+                logging.debug("\tfound {} P sub-subsection"
+                              .format(len(rsec["P"])))
+                for (l, l1), psec in rsec["P"].items():
+                    logging.debug("\treading sub-subsection for (P{},P{})"
+                                  .format(l, l1))
+                    logging.debug("\tfound {} NI-type sub-sub-subsection"
+                                  .format(len(psec["NI"])))
+                    covs = []
+                    for i, nisec in psec["NI"].items():
+                        logging.debug("\t\treconstruct covariance from NI-type section LB={}"
+                                      .format(nisec["LB"]))
+                        if nisec["LB"] == 5:
+                            foo = sandy.EnergyCov.from_lb5_asym if nisec["LS"] == 0 else sandy.EnergyCov.from_lb5_sym
+                            cov = foo(nisec["EK"], nisec["FKK"])
+                            covs.append(cov)
+                        elif nisec["LB"] == 1:
+                            cov = sandy.EnergyCov.from_lb1(nisec["EK"],
+                                                           nisec["FK"])
+                            covs.append(cov)
+                        elif nisec["LB"] == 2:
+                            cov = sandy.EnergyCov.from_lb2(nisec["EK"],
+                                                           nisec["FK"])
+                            covs.append(cov)
+                        elif nisec["LB"] == 6:
+                            cov = sandy.EnergyCov.from_lb6(nisec["EK"],
+                                                           nisec["EL"],
+                                                           nisec["FKL"])
+                            covs.append(cov)
+                        else:
+                            logging.warn("skip LB={} covariance for [({}/{}), ({}/{})]"
+                                         .format(nisec["LB"], mat, mt, mat1, mt1))
+                            continue
+                    if len(covs) == 0:
+                        logging.debug("\tsubsection MAT1={}/MT1={} did not provide accetable covariances"
+                                      .format(mat1, mt1))
+                        continue
+                    cov = sandy.EnergyCov.sum_covs(*covs)
+                    if cov.data.all().all():
+                        logging.warn("\tempty covariance for [({}/{}), ({}/{})]"
+                                     .format(mat, mt, mat1, mt1))
+                        continue
+                    index = pd.MultiIndex.from_product(
+                        [[mat], [mt], [l], cov.data.index],
+                        names=["MAT", "MT", "L", "E"],
+                        )
+                    columns = pd.MultiIndex.from_product(
+                        [[mat], [mt], [l], cov.data.index],
```

Comment:
this should be `mat1`, `mt1`, `l1`...it is not clear. primarily because it is fully untested